tls/s2n_signature_algorithms.c

```diff
@@ -154,19 +154,6 @@ int s2n_choose_sig_scheme_from_peer_preference_list(struct s2n_connection *conn,
         GUARD(s2n_choose_default_sig_scheme(conn, &chosen_scheme));
     }
 
-    /* In TLS 1.3, SigScheme also defines the ECDSA curve to use (instead of reusing whatever ECDHE Key Exchange curve
-     * was negotiated). In TLS 1.2 and before, chosen_scheme.signature_curve must *always* be NULL, if it's not, it's
-     * a bug in s2n's preference list. */
-    S2N_ERROR_IF(conn->actual_protocol_version <= S2N_TLS12 && chosen_scheme.signature_curve != NULL, S2N_ERR_INVALID_SIGNATURE_SCHEME);
```

Comment:
if this happens where else would catch/reflect the error?

Reply:
i think by adding the maximum protocol version to the definitions, it will now get checked in `s2n_choose_sig_scheme` which will fail on line 151 if there are no valid schemes.